articles/step-by-step-on-how-to-user-nodejs-repl-tutorial/index.md

```diff
@@ -0,0 +1,203 @@
+### Introduction
+Node.js `Read -Evaluate-Print-Loop` is an easy-to-use command-line tool for processing Node.js expressions.
+This shell captures the user's JavaScript code inputs, interprets and evaluates the result of this code, displays the result to the screen, and repeats the process till the user quits the shell.  
+
+It's therefore, important to note that this tool does not require file creation to write codes. It comes with ready to use Node.js development environment.
+
+### Prerequisites
+This tutorial requires:
+1. Node.js Installed in your development local machine.
+2. Basic knowledge of JavaScript.
+
+### Getting started with REPL
+Enter the following in the terminal:
+
+```bash
+$ node
+
+```
+Output:
+
+```bash
+> 
+```
+Entering the command `node` in the terminal starts the REPL command-line tool indicated by the `>` symbol.  
+This symbol acts as an indicator that JavaScript is ready to read and evaluate your code.  
+
+We can test the functionality by printing 'hello world' using REPL as shown below:
+
+```bash
+$ node
+> let hello = 'hello world';
+undefined
+> hello
+'hello world'  // output
+> 
+
+```
+In the script, we declare a variable `hello` and assign the text 'hello world'.  
+Next, we call this variable and press enter, to print the 'hello world' string on the screen. That's it.  
+
+Now how do we exit REPL?
+To exit this tool, press `ctrl + Z on your keyboard.  
+
+Output:
+```bash
+> let hello = 'hello world';
+undefined
+> hello
+'hello world'
+> 
+(To exit, press Ctrl+C again or Ctrl+D or type .exit)
+> 
+
+```
+As shown in the output, you can as well use `ctrl+D` or type `.exit`.  
+
+### Executing code using REPL.
+
+As discussed previously, this tool simplifies work by providing a quick way to test Node.js code without creating files.  
+> It's also important to note that any valid JavaScript code can be executed using REPL.
+
+ 1. Performing Arithmetical operations in REPL
+In the previous example, we printed 'hello world' string on the screen, in this section let's dive in and look at arithmetics:  
+
+In your terminal, start REPL: 
+
+```bash
+$ node
+> 
+
+```
+Let's perform basic addition, subtraction,modulus, division and multipliation respectively:
+Output:  
+
+```bash
+> 2+2
+4
+> 5-2
+3
+> 10%3
+1
+> 20/2
+10
+> 10*2
+20
+> 
+
+```
+2. Performing operations using Node's Math library
+Math library is an object with defined methods and properties. We can therefore use these features to perform some arithmetics.  
+
+Example:
+
+```bash
+---skipped lines
+
+> function getRandomNumber(maximum){
+... return Math.floor(Math.random()*maximum);
+... }
+undefined
```

Comment:
This snippet is unclear. Break it down into 2 different code snippets - one for input and other for output

Reply:
This is a simple Node.js method, returning a random number, how is it unclear?

Reply:
@odiwuoramos 
What I meant by unclear is - 

```bash
function getRandomNumber(maximum){
... return Math.floor(Math.random()*maximum);
... }
```
The above snippet is more like an input, and you get the output as shown below:

```bash
undefined
```

Now, you had put them under same code snippet. So, there are high chances that the reader might understand writing `undefined` as an extension to the above function.

That's the reason, I suggest to put them under different snippets like:

```bash
> function getRandomNumber(maximum){
... return Math.floor(Math.random()*maximum);
... }
```

**Output:**

```bash
undefined
```

Or better put a screenshot, it would resolve all the confusions.

Hope it is clear.